cmake/modules/CheckCxxAtomic.cmake

```diff
@@ -30,6 +30,7 @@ cmake_push_check_state()
 check_cxx_atomics(HAVE_CXX11_ATOMIC)
 cmake_pop_check_state()
 
+set(HAVE_CXX11_ATOMIC FALSE)
```

Comment:
i'd suggest check the assembly of the program compiled and linked above on s390x.

Reply:
Thanks, @tchaikov. The last time I looked at assembly code was . . . a very long time ago. For me that would be a long journey, which I'm reluctant to embark on without knowing how it would help fix the s390x build of master?

Reply:
then, that's probably not worthy doing this. i just wanted to understand if `__atomic_load_16` is used in that compiled test program or not.

Reply:
I uploaded the assembly to the ticket. I can see the following line:

```
        brasl   %r14,__atomic_load_16
```

Reply:
But then the test ought to fail ...

Can you run the compile of the test program without -S and see whether you get an error then?

If not, it would be good if you could run with -v and paste the output this shows.  Thanks!

Reply:
And, as I suspected, the following succeeds:

```
smithfarm@s390vsl019:~> /usr/bin/c++ -DHAVE_CXX11_ATOMIC -std=c++11 -latomic test.c
smithfarm@s390vsl019:~> 
```

Reply:
@uweigand I'm really sorry - I inadvertently replied to your comment by editing it instead of creating my own. Please forgive this stupid mistake of mine! Often my fingers are faster than my brain!!

Reply:
@uweigand I can confirm that compilation without `-S` results in an error:

```
smithfarm@s390vsl019:~> /usr/bin/c++ -DHAVE_CXX11_ATOMIC -std=c++11 test.c
/usr/lib64/gcc/s390x-suse-linux/7/../../../../s390x-suse-linux/bin/ld: /tmp/ccIfR54F.o: in function `std::atomic<unsigned __int128>::load(std::memory_order) const':
test.c:(.text._ZNKSt6atomicIoE4loadESt12memory_order[_ZNKSt6atomicIoE4loadESt12memory_order]+0x5e): undefined reference to `__atomic_load_16'
collect2: error: ld returned 1 exit status
smithfarm@s390vsl019:~>
```

Reply:
But that's all as expected then, and everything should just work.  If linking without -latomic fails, then the cmake test is supposed to notice that and therefore not set HAVE_CXX11_ATOMIC.

So either there's something wrong with cmake here (to verify this, it would be good to look at the CMakeOutput.log / CMakeError.log generated during the cmake run), or there is some difference between the system you ran this test on, and the system where the actual ceph build happened.